src/helper.jl

```diff
@@ -165,3 +165,230 @@ function writetable(filename::AbstractString; rewrite::Bool=false, kw...)
     writexlsx(filename, xf, rewrite=rewrite)
     nothing
 end
+
+"""
+    emptyfile(sheetname::AbstractString="")
+
+Returns an empty, writable `XLSXFile` with 1 worksheet.
+
+`sheetname` is the name of the worksheet, defaults to `Sheet1`.
+"""
+function emptyfile(sheetname::AbstractString="")
+    xf = open_default_template()
+
+    if sheetname != ""
+        rename!(xf[1], sheetname)
+    end
+
+    return xf
+end
+
+"""
+    openxlsx(f::Function, filename::AbstractString, sheetname::AbstractString=""; kw...)
+
+Open XLSX file for reading and/or writing.
+
+`filename` is the name of the file.
+`sheetname` is the name of the worksheet for a new file.
+
+If `rewrite=true` the file `filename` will overwritten is it exists.
+
+If `enable_cache=true`, all read worksheet cells will be cached.
+If you read a worksheet cell twice it will use the cached value instead of reading from disk
+in the second time.
+
+If `enable_cache=false`, worksheet cells will always be read from disk.
+This is useful when you want to read a spreadsheet that doesn't fit into memory.
+
+The default value is `enable_cache=true`.
+
+Example:
+
+```julia
+XLSX.openxlsx("new.xlsx", "New Sheet") do xf
+    sheet = xf["New Sheet"]
+    sheet[1, :] = [1, Date(2018, 1, 1), "test"]
+end
+
+XLSX.openxlsx("new.xlsx") do xf
+    sheet = xf["New Sheet"]
+    data = sheet[:]
+end
+
+XLSX.openxlsx("new.xlsx", "New Sheet 2", rewrite=true) do xf
+    sheet = xf["New Sheet"]
+    sheet[1, 1] = "New data"
+end
+```
+"""
+function openxlsx(f::Function, filename::AbstractString, sheetname::AbstractString=""; rewrite::Bool=false, enable_cache::Bool=true)
+    writefile = false
+
+    if isfile(filename) && !rewrite
+        xf = open_or_read_xlsx(filename, false, enable_cache, false)
+    else
+        writefile = true
+        xf = open_default_template()
+
+        if sheetname != ""
+            rename!(xf[1], sheetname)
+        end
```

Comment:
I think we don't need to interact with sheets at all in this method and can remove `sheetname` as a parameter